src/lib/flatten.js

```diff
@@ -0,0 +1,21 @@
+const flatten = (startingArray) => {
+  var newArray = [];
```

Comment:
I recommend making use of `let` and `const` in ES6 to avoid some possible declaration bugs (though your declaration at the top of the function scope tends to avoid those bugs).  [Here's a good article](https://medium.com/javascript-scene/javascript-es6-var-let-or-const-ba58b8dcde75#.kx8qya1aa) that discusses some of the reasoning behind this recommendation.